src/beast/beast/wsproto/rfc6455.h

```diff
@@ -0,0 +1,154 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of Beast: https://github.com/vinniefalco/Beast
+    Copyright 2013, Vinnie Falco <vinnie.falco@gmail.com>
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#ifndef BEAST_WSPROTO_RFC6455_H_INCLUDED
+#define BEAST_WSPROTO_RFC6455_H_INCLUDED
+
+#include <beast/wsproto/static_string.h>
+#include <boost/optional.hpp>
+#include <array>
+#include <cstdint>
+
+namespace beast {
+namespace wsproto {
+
+/** WebSocket frame header opcodes. */
+namespace opcode {
+    enum value : std::uint8_t
+    {
+        cont    = 0,
+        text    = 1,
+        binary  = 2,
+        rsv3    = 3,
+        rsv4    = 4,
+        rsv5    = 5,
+        rsv6    = 6,
+        rsv7    = 7,
+        close   = 8,
+        ping    = 9,
+        pong    = 10,
+        crsvb   = 11,
+        crsvc   = 12,
+        crsvd   = 13,
+        crsve   = 14,
+        crsvf   = 15
+    };
+}
```

Comment:
I believe we should strongly prefer `enum class` over a namespaced `enum` whenever it is practical to do so.  Here is `opcode::value` converted to `opcode`:  https://github.com/HowardHinnant/rippled/commit/751cc455a5d30c7b93aa3a27c5e8982746ba23cb